pkg/generator/types.go

```diff
@@ -185,9 +186,65 @@ func varNotes(variable broker.BrokerVariable) string {
 		out += fmt.Sprintf(" Default: `%v`.", variable.Default)
 	}
 
+	bullets := constraintsToDoc(variable.ToSchema())
+	if len(bullets) > 0 {
+		out += "\n    * "
+		out += strings.Join(bullets, "\n    * ")
+	}
+
 	return out
 }
 
+// constraintsToDoc converts a map of JSON Schema validation key/values to human-readable bullet points.
+func constraintsToDoc(schema map[string]interface{}) []string {
+	// We use an anonymous struct rather than a map to get a strict ordering of
+	// constraints so they are generated consistently in documentation.
+	// Not all JSON Schema constraints can be cleanly expressed in this format,
+	// nor do we use them all so some are missing.
+	constraintFormatters := []struct {
+		SchemaKey string
+		DocString string
+	}{
+		// Schema Annotations
+		{validation.KeyExamples, "Examples: %+v."},
+
+		// Validation for any instance type
+		{validation.KeyEnum, "The value must be one of: %+v."},
+		{validation.KeyConst, "The value must be: `%v`."},
+
+		// Validation keywords for numeric instances
+		{validation.KeyMultipleOf, "The value must be a multiple of %v."},
+		{validation.KeyMaximum, "The value must be less than or equal to %v."},
+		{validation.KeyExclusiveMaximum, "The value must be strictly less than %v."},
+		{validation.KeyMinimum, "The value must be greater than or equal to %v."},
+		{validation.KeyExclusiveMaximum, "The value must be strictly greater than %v."},
+
+		// Validation keywords for strings
+		{validation.KeyMaxLength, "The string must have at most %v characters."},
+		{validation.KeyMinLength, "The string must have at least %v characters."},
+		{validation.KeyPattern, "The string must match the regular expression `%v`."},
+
+		// Validation keywords for arrays
+		{validation.KeyMaxItems, "The array must have at most %v items."},
+		{validation.KeyMinItems, "The array must have at least %v items."},
+
+		// Validation keywords for objects
+		{validation.KeyMaxProperties, "The object must have at most %v properties."},
+		{validation.KeyMinProperties, "The object must have at least %v properties."},
+		{validation.KeyRequired, "The following properties are required: %v."},
+		{validation.KeyPropertyNames, "Property names must match the JSON Schema: `%+v`."},
+	}
+
+	var bullets []string
+	for _, formatter := range constraintFormatters {
+		if v, ok := schema[formatter.SchemaKey]; ok {
```

Comment:
Should we error or log on `!ok`? Just in case someone adds a constraint but forgets to add a new bullet?

Reply:
I think it's a risk we'll have to take. There are some fields we don't want to document in this format (like name, type, and details which are accounted for in the section above the bullets) and I doubt the spec will change much, but it might. 

I left a comment up in the struct for us in the future to remember why not everything gets a bullet point.